xchange-bitfinex/src/main/java/org/knowm/xchange/bitfinex/service/BitfinexMarketDataServiceRaw.java

```diff
@@ -77,15 +78,23 @@ public BitfinexLendDepth getBitfinexLendBook(String currency, int limitBids, int
   }
 
   public List<CurrencyPair> getExchangeSymbols() throws IOException {
-    List<CurrencyPair> currencyPairs = new ArrayList<>();
-    for (String symbol : bitfinex.getSymbols()) {
-      currencyPairs.add(BitfinexAdapters.adaptCurrencyPair(symbol));
+    try {
+      List<CurrencyPair> currencyPairs = new ArrayList<>();
+      for (String symbol : bitfinex.getSymbols()) {
+        currencyPairs.add(BitfinexAdapters.adaptCurrencyPair(symbol));
+      }
+      return currencyPairs;
+    } catch (BitfinexException e) {
```

Comment:
this type of mapping should be done in non-Raw *Service classes
Raw *Service classes should throw exchange native exceptions

see: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices#step-7